src/main/java/com/launchdarkly/eventsource/EventSource.java

```diff
@@ -23,18 +20,18 @@
 
   private final URI uri;
   private final Headers headers;
-  private final ExecutorService executor;
+  private final ExecutorService executor = Executors.newCachedThreadPool();
```

Comment:
It's very minor, but ensures the executor is never null (ie if another constructor is added or a subclass does something weird) which means we can avoid a null check in the stop() method. 
Also, it makes the code more readable. Navigating to its declaration tells you everything you need to know about it's initial state.